### PR TITLE
Add cloudformation support for UserContext

### PR DESCRIPTION
**Pull request by ishaan-berry**
*Issue #, if available:*
Add support to create kendra index with user context configurations.
*Description of changes:*
Added support to create/update/read kendra index with user context configurations via cloudformation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java [hunk 1]
```diff
@@ -123,6 +131,8 @@ static ResourceModel translateFromReadResponse(final DescribeIndexResponse descr
                 .queryCapacityUnits(describeIndexResponse.capacityUnits().queryCapacityUnits())
                 .build());
       }
+    } if (describeIndexResponse.hasUserTokenConfigurations()) {
```
Comment (angranamzn) [inline aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java:134]: np; new line

File: aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java [hunk 2]
```diff
@@ -162,7 +173,11 @@ static UpdateIndexRequest translateToUpdateRequest(final ResourceModel currModel
                             currModel.getDocumentMetadataConfigurations(),
                             prevDocumentMetadataConfiguration))
             .capacityUnits(translateToCapacityUnitsConfiguration(currModel.getCapacityUnits(), currModel.getEdition()))
-            .build();
+            .userTokenConfigurations(translateToSdkUserTokenConfigurations(currModel.getUserTokenConfigurations()));
+    if (currModel.getUserContextPolicy() != null) {
+      builder.userContextPolicy(currModel.getUserContextPolicy());
```
Comment (angranamzn) [inline aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java:178]: np; do we need this if statement? Can we just have `.userContextPolicy(currModel.getUserContextPolicy())` above? If we can get rid of the check we wont have to worry about an extra branch to test (maybe we happen to cover both branches now but in the future that could change).

File: aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java (hunk 1 above)
Comment (angranamzn) [inline aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java:134]: Do you need this check? Looks like `translateFromSdkUserTokenConfigurations` does the check for you - I'd just have it in one place. Another reason I'd remove the check is having it results in code coverage going down unless we have tests that check both branches.

File: aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java [hunk 3]
```diff
@@ -55,6 +59,9 @@ static CreateIndexRequest translateToCreateRequest(final ResourceModel model) {
                       .kmsKeyId(model.getServerSideEncryptionConfiguration().getKmsKeyId())
                       .build());
     }
+    if (model.getUserContextPolicy() != null) {
```
Comment (angranamzn) [inline aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java:62]: Do we need this if statement?

File: aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java (hunk 2 above)
Comment (ishaan-berry) [reply aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java:178]: Updated in :https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kendra/pull/81

GitHubGitHub

File: aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java (hunk 1 above)
Comment (ishaan-berry) [reply aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java:134]: Updated in : https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kendra/pull/81

GitHubGitHub

File: aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java (hunk 3 above)
Comment (ishaan-berry) [reply aws-kendra-index/src/main/java/software/amazon/kendra/index/Translator.java:62]: Removed

**Review by EricCeres — COMMENTED** [review]
LGTM.

**Pull request closed without merging**